std/hl/_std/Reflect.hx

```diff
@@ -32,6 +32,16 @@ class Reflect {
 		if (field == null)
 			return null;
 		var hash = @:privateAccess field.bytes.hash();
+		if (Std.isOfType(o, hl.BaseType.Enum)) {
```

Comment:
I'm worried about a generic is-check like this because it could potentially browse deep class hierarchies in search for the Enum type, which it is never going to find in most cases...

Reply:
Switched to generating fields on the enum class instead, which should avoid this issue.